src/popover/popover.directive.ts

```diff
@@ -84,31 +95,172 @@ export class PopoverContainer {
+	 * Handles emitting open/close event
+	 * @param open - Is the popover container open
+	 * @param event - Event
+	 */
+	handleChange(open: boolean, event?: Event) {
+		if (this._open !== open) {
 			this.isOpenChange.emit(open);
 		}
 
 		if (open) {
-			this.onOpen.emit(event);
+			if (event) {
+				this.onOpen.emit(event);
+			}
+
+			// auto alignment is enabled, we use auto update to set the placement for the element
+			if (this.autoAlign) {
+				if (this.caretRef) {
+					// Get caret offset/height property
+					this.caretOffset = parseFloat(
+						getComputedStyle(this.caretRef).getPropertyValue("--cds-popover-offset").split("rem", 1)[0]
+					) * 16 || 10;
+					this.caretHeight = parseFloat(
+						getComputedStyle(this.caretRef).getPropertyValue("--cds-popover-caret-height").split("px", 1)[0]
+					) || 6;
+				}
+				if (this.elementRef.nativeElement && this.popoverContentRef) {
+					// Auto update
+					this.autoUpdateCleanUp = autoUpdate(this.elementRef.nativeElement, this.popoverContentRef, this.recomputePosition.bind(this));
+				}
+			}
+
 		} else {
-			this.onClose.emit(event);
+			this.cleanUp();
+			if (event) {
+				this.onClose.emit(event);
+			}
 		}
-		this.isOpen = open;
+		this._open = open;
 		this.changeDetectorRef.markForCheck();
 	}
+
+	/**
+	 * Compute position of tooltip when autoAlign is enabled
+	 */
+	async recomputePosition() {
+		// Run outside of angular zone to avoid change detection and rely on
+		// floating-ui
+		this.ngZone.runOutsideAngular(async () => {
+			const { x, y, placement, middlewareData } = await computePosition(
+				this.elementRef.nativeElement,
+				this.popoverContentRef,
+				{
+					placement: this._align,
+					strategy: "fixed",
+					middleware: [
+						flip({ fallbackAxisSideDirection: "start" }),
+						offset(this.caretOffset),
+						this.caret && arrow({ element: this.caretRef })
+					]
+				});
+
+			const previousAlignment = this._align;
+			this._align = placement;
+			this.setAlignmentClass(previousAlignment);
+
+			// Using CSSOM to manipulate CSS to avoid content security policy inline-src
+			// https://github.com/w3c/webappsec-csp/issues/212
+			this.popoverContentRef.style.position = "fixed";
+			this.popoverContentRef.style.left = `${x}px`;
+			this.popoverContentRef.style.top = `${y}px`;
+
+			if (middlewareData.arrow) {
+				const { x: arrowX, y: arrowY } = middlewareData.arrow;
+
+				const staticSide = {
+					top: "bottom",
+					right: "left",
+					bottom: "top",
+					left: "right"
+				}[placement.split("-")[0]];
+
+				this.caretRef.style.left = arrowX != null ? `${arrowX}px` : "";
+				this.caretRef.style.top = arrowY != null ? `${arrowY}px` : "";
+				this.caretRef.style.right = "";
+				this.caretRef.style.bottom = "";
+				this.caretRef.style[staticSide] = `${-this.caretHeight}px`;
+			}
+		});
+	}
+
+	/**
+	 * Handle initialization of element
+	 */
+	ngAfterViewInit(): void {
+		this.initialzeReferences();
+	}
+
+	initialzeReferences(): void {
```

Comment:
Small typo in this function name `initializeReferences` vs `initialzeReferences`